packages/shared/src/components/widgets/PostUsersHighlights.tsx

```diff
@@ -31,18 +33,16 @@ enum UserType {
 
 const StyledImage = classed(LazyImage, 'w-10 h-10');
 
-interface SourceAuthorProps {
-  id?: string;
-  handle?: string;
-  image: string;
-  name: string;
-  username?: string;
-  userType?: UserType;
-  permalink: string;
-  reputation?: number;
-}
+type SourceAuthorProps =
+  | ({ userType?: null } & CommentAuthor)
+  | ({
+      userType: UserType.Featured | UserType.Author | UserType.Scout;
+    } & CommentAuthor)
+  | ({
+      userType: UserType.Source;
+    } & ISource);
 
-interface UserHighlightProps extends SourceAuthorProps {
+type UserHighlightProps = SourceAuthorProps & {
```

Comment:
As Chris said, if there is no meaningful need for this change let's remove since it adds a mental overload when looking at changed code in the PR.

Reply:
@capJavert I need this change because `block` action need more than the `id` and there is no way to have this data not changing this prop and how it behave :(

Reply:
but the `interface` to `type` change is not needed right?

Reply:
Missed this comment from all the changes https://github.com/dailydotdev/apps/pull/3360/files#r1682862007, my mistake.